### ISSUE TITLE
caffe 训练数据集

### ISSUE THREAD

**Issue by yuqingbin** [njvisionpower/Safety-Helmet-Wearing-Dataset#6]
您好，我还有个问题咨询下，您提供的数据集我可以直接拿来在caffe-ssd框架下训练吗？我看caffe VOC0712 数据集里面 ImageSets  Main, txt 文件里名称后面有-1 标注，我尝试直接用VOC2028 ，create_list.sh 都没成功，不知是否我使用不当

**Comment by njvisionpower** [comment]
可以贴一下报错的log吗？每个框架或者方法预处理Pascal voc数据集格式不太一样。

**Comment by yuqingbin** [comment]
> 可以贴一下报错的log吗？每个框架或者方法预处理Pascal voc数据集格式不太一样。

Create list for VOC2028 trainval...
Create list for VOC2028 test...
I0925 09:08:59.214675  9505 get_image_size.cpp:61] A total of 3034 images.
E0925 09:08:59.215683  9505 io.cpp:187] Could not open or find file /home/yuqingbin/data/VOCdevkit/VOC2028/JPEGImages/000002
E0925 09:08:59.216178  9505 io.cpp:187] Could not open or find file /home/yuqingbin/data/VOCdevkit/VOC2028/Annotations/000002
E0925 09:08:59.216819  9505 io.cpp:187] Could not open or find file /home/yuqingbin/data/VOCdevkit/VOC2028/JPEGImages/000005
E0925 09:08:59.216900  9505 io.cpp:187] Could not open or find file /home/yuqingbin/data/VOCdevkit/VOC2028/Annotations/000005
E0925 09:08:59.216976  9505 io.cpp:187] Could not open or find file /home/yuqingbin/data/VOCdevkit/VOC2028/JPEGImages/000019
E0925 0

**Comment by yuqingbin** [comment]
ubuntu 19,路径我去看了一下文件是存在的

**Comment by njvisionpower** [comment]
可以参考这篇[blog](https://blog.csdn.net/10km/article/details/70144925)，我的数据集是在Windows下标注的，所以Linux下转的时候可能存在字符不匹配的问题。

**Comment by yuqingbin** [comment]
> 可以参考这篇[blog](https://blog.csdn.net/10km/article/details/70144925)，我的数据集是在Windows下标注的，所以Linux下转的时候可能存在字符不匹配的问题。

好的，应该是这个问题

**Comment by yuqingbin** [comment]
训练时出现如下报错，我是用的CPU

I0925 13:27:05.095297  5783 net.cpp:761] Ignoring source layer prob
I0925 13:27:05.097661  5783 caffe.cpp:251] Starting Optimization
I0925 13:27:05.097668  5783 solver.cpp:294] Solving VGG_VOC2028_SSD_300x300_train
I0925 13:27:05.097671  5783 solver.cpp:295] Learning Rate Policy: multistep
I0925 13:27:05.124804  5783 blocking_queue.cpp:50] Data layer prefetch queue empty
I0925 13:27:59.117923  5783 solver.cpp:243] Iteration 0, loss = 17.7011
I0925 13:27:59.118057  5783 solver.cpp:259]     Train net output #0: mbox_loss = 17.7011 (* 1 = 17.7011 loss)
I0925 13:27:59.118083  5783 sgd_solver.cpp:138] Iteration 0, lr = 0.001

terminate called after throwing an instance of 'cv::Exception'
  what():  OpenCV(3.4.7) /home/yuqingbin/AI/opencv-3.4.7/modules/imgproc/src/color.simd_helpers.hpp:88: error: (-2:Unspecified error) in function 'cv::impl::{anonymous}::CvtHelper<VScn, VDcn, VDepth, sizePolicy>::CvtHelper(cv::InputArray, cv::OutputArray, int) [with VScn = cv::impl::{anonymous}::Set<3, 4>; VDcn = cv::impl::{anonymous}::Set<3>; VDepth = cv::impl::{anonymous}::Set<0, 5>; cv::impl::{anonymous}::SizePolicy sizePolicy = cv::impl::<unnamed>::NONE; cv::InputArray = const cv::_InputArray&; cv::OutputArray = const cv::_OutputArray&]'
> Invalid number of channels in input image:
>     'VScn::contains(scn)'
> where
>     'scn' is 1

*** Aborted at 1569389531 (unix time) try "date -d @1569389531" if you are using GNU date ***
PC: @     0x7f40766243eb gsignal
*** SIGABRT (@0x3e800001697) received by PID 5783 (TID 0x7f404885d700) from PID 5783; stack trace: ***
    @     0x7f4076cf7641 (unknown)
    @     0x7f4076624470 (unknown)
    @     0x7f40766243eb gsignal
    @     0x7f4076603899 abort
    @     0x7f40768895f6 (unknown)
    @     0x7f40768959ec (unknown)
    @     0x7f4076895a47 std::terminate()
    @     0x7f4076895ca9 __cxa_throw
    @     0x7f407506dce0 _ZN2cv5errorERKNS_9ExceptionE.cold
    @     0x7f40752b1003 cv::error()
    @     0x7f40750c3d4c cv::errorNoReturn()
    @     0x7f40750c53d7 cv::detail::check_failed_auto_<>()
    @     0x7f40750c5418 cv::detail::check_failed_auto()
    @     0x7f407230b114 cv::impl::(anonymous namespace)::CvtHelper<>::CvtHelper()
    @     0x7f4072310120 cv::cvtColorBGR2HSV()
    @     0x7f4072303618 cv::cvtColor()
    @     0x7f4076f89a75 caffe::AdjustSaturation()
    @     0x7f4076f8aa20 caffe::RandomSaturation()
    @     0x7f4076f8b20f caffe::ApplyDistort()
    @     0x7f4076fac056 caffe::DataTransformer<>::DistortImage()
    @     0x7f407712f237 caffe::AnnotatedDataLayer<>::load_batch()
    @     0x7f407704ab6f caffe::BasePrefetchingDataLayer<>::InternalThreadEntry()
    @     0x7f40719371b5 (unknown)
    @     0x7f406f5d5669 start_thread
    @     0x7f40767002f3 clone
Aborted (core dumped)

**Comment by yuqingbin** [comment]
图片通道问题？

**Comment by njvisionpower** [comment]
输入通道的问题，看起来数据增强部分图像输入的通道是1维(灰度图像？)但转的时候是从BGR转的，所以不匹配，可以看下这个的[解释](https://github.com/aleju/imgaug/issues/157)，可能是opencv imread 的时候flag的问题

**Comment by yuqingbin** [comment]
https://github.com/weiliu89/caffe/blob/ssd/examples/ssd/ssd_coco.py#L178

谢谢，已解决
